addons/website_sale_wallet/models/sale_order.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    # Relational fields
+    generated_gift_card_ids = fields.One2many('gift.card', "sale_order_purchase_id",
+                                              string="Bought Gift Card",
+                                              copy=False)
+    _wallet_product_id = fields.Many2one('product.product', store=False)
+    order_line_wallet_id = fields.Many2one('sale.order.line')
+
+    # Compute fields
+    generated_gift_card_count = fields.Integer(compute="_compute_gift_card_purchased", default=0)
+    wallet_partner_balance = fields.Monetary(related='partner_id.balance')
+    wallet_partner_actual_balance = fields.Monetary(related='partner_id.actual_balance')
+
+    # Compute Func
+    @api.constrains('state')
+    def _compute_purchase_gift_card(self):
```

Comment:
compute name in method name is wrong idea since it is not a compute method for a field